Noise Estimator Bug

There appears to be a long standing bug in the noise estimator code.  One of the post docs reviewing the code against their own implementation pointed it out.  

`np.abs(x**2)` vs `np.abs(x)**2`

For the white noise estimator case, the bug doesn't change any values, since the values are real dtypes.  For AnIso case, the values are complex, and it does change results.

I took a look at some of the other places we use that calculation (adaptive support, reddy_chatterji) and those are using the expected np.abs(x)**2.

Its a small code change, putting patch together now.

I think I agree on paper, both resulting `a**2 + b **2`.  However, checking the actualcalculations indicated they are not `allclose` in practice.  They are close enough that I have no worries about the (psd) calculations :).  So given the two formulas I prefer the one implemented here. Maybe bug was too strong a word ... but it was enough to confuse our current researchers who asked me to check it.... It should now be the same across our code and papers; at least in the places I found to check.

```
diff --git a/src/aspire/noise/noise.py b/src/aspire/noise/noise.py
index 88e03caf..384c08f4 100644
--- a/src/aspire/noise/noise.py
+++ b/src/aspire/noise/noise.py
@@ -331,6 +331,7 @@ class AnisotropicNoiseEstimator(NoiseEstimator):
 
         mean_est = 0
         noise_psd_est = np.zeros((self.src.L, self.src.L)).astype(self.src.dtype)
+        noise_psd_est_old = np.copy(noise_psd_est)
         for i in trange(0, self.src.n, self.batchSize):
             images = self.src.images[i : i + self.batchSize].asnumpy()
             images_masked = images * mask
@@ -339,6 +340,9 @@ class AnisotropicNoiseEstimator(NoiseEstimator):
             mean_est += np.sum(images_masked) / _denominator
             im_masked_f = xp.asnumpy(fft.centered_fft2(xp.asarray(images_masked)))
             noise_psd_est += np.sum(np.abs(im_masked_f) ** 2, axis=0) / _denominator
+            noise_psd_est_old += np.sum(np.abs(im_masked_f ** 2), axis=0) / _denominator
+
+        np.testing.assert_allclose(noise_psd_est_old, noise_psd_est)
 
         mid = self.src.L // 2
         noise_psd_est[mid, mid] -= mean_est**2
(END)
```

...
```
E           AssertionError: 
E           Not equal to tolerance rtol=1e-07, atol=0
E           
E           Mismatched elements: 549 / 4225 (13%)
E           Max absolute difference: 1.1920929e-07
E           Max relative difference: 3.2296424e-07
E            x: array([[0.046381, 0.044574, 0.041904, ..., 0.042834, 0.045083, 0.046031],
E                  [0.048914, 0.045155, 0.048245, ..., 0.049979, 0.046872, 0.047778],
E                  [0.048787, 0.044889, 0.053858, ..., 0.048414, 0.047677, 0.048151],...
E            y: array([[0.046381, 0.044574, 0.041904, ..., 0.042834, 0.045083, 0.046031],
E                  [0.048914, 0.045155, 0.048245, ..., 0.049979, 0.046872, 0.047778],
E                  [0.048787, 0.044889, 0.053858, ..., 0.048414, 0.047677, 0.048151],...
```

--- a/src/aspire/noise/noise.py
+++ b/src/aspire/noise/noise.py
@@ -292,7 +292,7 @@ class WhiteNoiseEstimator(NoiseEstimator):
 
             _denominator = self.src.n * np.sum(mask)
             first_moment += np.sum(images_masked) / _denominator
-            second_moment += np.sum(np.abs(images_masked**2)) / _denominator
+            second_moment += np.sum(np.abs(images_masked) ** 2) / _denominator
         return second_moment - first_moment**2
 
 
@@ -338,7 +338,7 @@ class AnisotropicNoiseEstimator(NoiseEstimator):
             _denominator = self.src.n * np.sum(mask)
             mean_est += np.sum(images_masked) / _denominator
             im_masked_f = xp.asnumpy(fft.centered_fft2(xp.asarray(images_masked)))
-            noise_psd_est += np.sum(np.abs(im_masked_f**2), axis=0) / _denominator
+            noise_psd_est += np.sum(np.abs(im_masked_f) ** 2, axis=0) / _denominator
 
         mid = self.src.L // 2
         noise_psd_est[mid, mid] -= mean_est**2

